From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Christian=20G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Sat, 27 Dec 2025 15:39:11 +0100
Subject: [PATCH] libselinux/src/Makefile: build python module without
 isolation

The "pip install" logic in libselinux/src/Makefile tries to get
setuptools from the network, which is not correct as during the
build/installation, there shouldn't be any network access, or
downloading of arbitrary resources from the network.

This causes build issues in Debian:

  https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1121121

And in Buildroot:

  https://autobuild.buildroot.net/results/0e9/0e9de0c0d8b6ec57eea9f8834f02076b296ba4f1/build-end.log

To fix this, we use the --no-build-isolation pip install argument so
that pip install is allowed to use the system-provided setuptools,
instead of wanting to download this one.

This patch is derived from
https://sources.debian.org/src/libselinux/3.9-4/debian/patches/Build-python-module-without-isolation-Closes-1119155.patch,
with the change that --no-build-isolation is passed unconditionally,
regardless of whether $(DESTDIR) is empty or not.

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Acked-by: James Carter <jwcart2@gmail.com>
---
 libselinux/src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libselinux/src/Makefile b/libselinux/src/Makefile
index 41cfbdca490cb55a7cf2d1fc0fbb6be3e89ce197..c5bd22cded125f2115e77e25d0da3eb7a0be1d86 100644
--- a/libselinux/src/Makefile
+++ b/libselinux/src/Makefile
@@ -191,7 +191,7 @@ install: all
 	ln -sf --relative $(DESTDIR)$(SHLIBDIR)/$(LIBSO) $(DESTDIR)$(LIBDIR)/$(TARGET)
 
 install-pywrap: pywrap
-	CFLAGS="$(CPPFLAGS) $(CFLAGS) $(SWIG_CFLAGS)" $(PYTHON) -m pip install --prefix=$(PREFIX) `test -n "$(DESTDIR)" && echo --root $(DESTDIR) --ignore-installed --no-deps` $(PYTHON_SETUP_ARGS) .
+	CFLAGS="$(CPPFLAGS) $(CFLAGS) $(SWIG_CFLAGS)" $(PYTHON) -m pip install --no-build-isolation --prefix=$(PREFIX) `test -n "$(DESTDIR)" && echo --root $(DESTDIR) --ignore-installed --no-deps` $(PYTHON_SETUP_ARGS) .
 	install -m 644 $(SWIGPYOUT) $(DESTDIR)$(PYTHONLIBDIR)/selinux/__init__.py
 	ln -sf --relative $(DESTDIR)$(PYTHONLIBDIR)/selinux/_selinux$(PYCEXT) $(DESTDIR)$(PYTHONLIBDIR)/_selinux$(PYCEXT)
 
